### PR TITLE
mqtt: implements last will and testament

### PR DESCRIPTION
**Pull request by pedrolamas**
Implements #262

While we discussed in the FR to have this under the "status" topic, I've gone with a separate "state" topic to ensure it is not overwritten by subsequent messages.

That way, one can check the "state" topic anytime to see if Moonraker is up and running!

Signed-off-by: Pedro Lamas <pedrolamas@gmail.com>

### CODE REVIEW

File: moonraker/components/mqtt.py [hunk 1]
```diff
@@ -191,6 +191,7 @@ def __init__(self, config: ConfigHelper) -> None:
         self.api_request_topic = f"{self.instance_name}/moonraker/api/request"
         self.api_resp_topic = f"{self.instance_name}/moonraker/api/response"
         self.klipper_status_topic = f"{self.instance_name}/klipper/status"
+        self.klipper_state_topic = f"{self.instance_name}/klipper/state"
```
Comment (pedrolamas) [inline moonraker/components/mqtt.py:194]: Come to think of it, I do wonder if this should be `{self.instance_name}/moonraker/state` or even just `{self.instance_name}/state`

**Comment by pedrolamas** [comment]
For the record, I've been using this locally with my printer and so far it seems to be working ok!

I've also confirmed that MQTT server will set to "offline" automatically if I use `kill -9 <pid>` on Moonraker process.

**Comment by Arksine** [comment]
Ok, from what I a gather you want to introduce a new topic that reports the server's state and set the last will and testament on that topic.  I think that this makes sense and adds value, however I do think that we need to change the topic.  Since the data is specific to Moonraker's status the topic should probably be `{instance_name}/moonraker/status`.  The payload could be a json object, initially set to `{"server": "online"}`, with the lwt set to `{"server": "offline"}`.  This gives us flexibility to expand what is reported through this topic in the future.

There doesn't appear to be a merge conflict with my recent changes, however it would probably be a good idea for you to rebase this branch against master.

**Comment by pedrolamas** [comment]
Excellent, I'll make the changes in a couple of hours when I'm back home!

**Comment by pedrolamas** [comment]
Changes we discussed are in now! 🙂

**Comment by pedrolamas** [comment]
@Arksine I did the change to JSON blob (`{"server": "online"}` and `{"server": "offline"}`) instead of a simple value (`online` and `offline`), but I think I should point that there seems to be a standard to use the simple value!

Here's an example of this, straight from Home Assistant documentation: https://www.home-assistant.io/integrations/sensor.mqtt/#availability_topic

**Comment by Arksine** [comment]
Looks good.  FWIW, I'm not sure I would consider that a standard.  It appears to be how Home Assistant implements its MQTT Sensor availability, but even there the payload is configurable.  I'm somewhat surprised that they didn't allow for the use of templates.  Regardless, I'm not sure it would be correct to configure Moonraker as a Sensor in Home Assistant.  Most likely it would need its own plugin.

**Comment by pedrolamas** [comment]
> I'm not sure it would be correct to configure Moonraker as a Sensor in Home Assistant

I've tried it already, it works fine and this small change for availability makes it even more powerful!! 😁

**Pull request merged**